boardswarm-cli/src/main.rs

```diff
@@ -242,6 +242,76 @@ enum AuthCommand {
     Modify(AuthModifyArg),
 }
 
+#[derive(Clone, Debug)]
+enum ItemArg {
+    Id(u64),
+    Name(String),
+}
+
+async fn item_lookup(
+    arg: ItemArg,
+    item_type: ItemType,
+    mut client: Boardswarm,
+) -> Result<u64, anyhow::Error> {
+    let item_kind = match item_type {
+        ItemType::Device => ("Device", "device"),
+        ItemType::Console => ("Console", "console"),
+        ItemType::Actuator => ("Actuator", "actuator"),
+        ItemType::Volume => ("Volume", "volume"),
+    };
+    match arg {
+        ItemArg::Id(id) => Ok(id),
+        ItemArg::Name(name) => {
+            let mut items = client.list(item_type).await?;
+
+            let name_components: Vec<_> = name.split('@').collect();
+            if name_components.len() > 2 {
+                return Err(anyhow!("Too many @ in {} name", item_kind.1));
+            }
+
```

Comment:
```suggestion
            let (name, instance) = name
                .rsplit_once('@')
                .map_or((name.as_str(), None), |(n, i)| (n, Some(i)));
```
no reason to create a vec and then do awkward indexing; you probably just want the name and an Option for the instance (as well it's optional)